src/nix/repl.cc

```diff
@@ -302,16 +336,25 @@ bool NixRepl::getLine(string & input, const std::string &prompt)
 
     if (!s)
       return false;
+#ifdef READLINE
+    if (*s) {
+        add_history(s);
+    }
+#endif
     input += s;
     input += '\n';
     return true;
 }
 
-
+#ifdef READLINE
+StringSet * NixRepl::completePrefix(string prefix)
+{
+    StringSet * completions = new StringSet;
```

Comment:
What is the reason for the function to not to return the same type in both cases?

Reply:
TBH I have no clue why had done it this way. I had made the changes a few months back. I have now cleaned up, the signature of the `completePrefix` function is now unchanged.